Mention the importance of HTTPS and risk of MitM attacks

noffle opened the pull request:
Though the example uses HTTPS in passing, none of the docs seem to explicitly mention why HTTPS is a Good Idea(tm) and the associated risks. This PR adds a few lines to the README talking about the why.

mstoiber commented:
Please add it to the Table of Contents, and if you could simplify the language a bit, that'd be awesome!

Thanks!

noffle commented:
I've added the entry to the Table of Contents, and fixed up some other formatting I had inadvertently broke.

Could you elaborate on what you mean by simplifying the language? What parts?

mstoiber commented:
Nevermind, I misread the paragraph, my bad. Merged into the main branch, thanks for adding that!
